vulnfeeds/pypi/pypi.go

```diff
@@ -134,25 +167,45 @@ func New(pypiLinksPath string, pypiVersionsPath string) *PyPI {
 	linksSource := loadLinks(pypiLinksPath)
 	versionsSource := loadVersions(pypiVersionsPath)
 
+	links, vendorProductToPkg := processLinks(linksSource)
 	return &PyPI{
-		links:           processLinks(linksSource),
-		versions:        processVersions(versionsSource),
-		checkedPackages: map[string]bool{},
+		links:              links,
+		versions:           processVersions(versionsSource),
+		checkedPackages:    map[string]bool{},
+		vendorProductToPkg: vendorProductToPkg,
 	}
 }
 
 func (p *PyPI) Matches(cve cves.CVEItem) string {
-	desc := cves.EnglishDescription(cve.CVE)
 	for _, reference := range cve.CVE.References.ReferenceData {
+		// If there is a PyPI link, it must be a Python package.
 		if pkg := extractPyPIProject(reference.URL); pkg != "" {
 			log.Printf("Matched via PyPI link: %s", reference.URL)
 			return pkg
 		}
 
-		if pkg := p.matchesPackage(reference.URL, desc); pkg != "" {
+		// Otherwise try to cross-reference the link against our set of known links.
+		if pkg := p.matchesPackage(reference.URL, cve.CVE); pkg != "" {
 			return pkg
 		}
 	}
+
+	// As a last resort, extract the vendor and product from the CPE and try to match that
+	// against vendor/product combinations extracted from e.g. GitHub links.
+	cpes := cves.CPEs(cve)
```

Comment:
Do you anticipate using this function elsewhere, or maybe you only need a single CPE?

Reply:
I want to keep the cves package general since that might be used by others, even if we don't use it anywhere else yet. 

A lot of this might end up also end up getting refactored later as we add support for more package ecosystems in the future to share more logic.